tests/unit/test_resolve_base_branch.py

```
@@ -0,0 +1,195 @@
+
+
+def test_auto_detect_main_only(git_repo: Path) -> None:
+    """main から切った feature branch は main を選ぶ。"""
+    _git(["checkout", "-q", "-b", "feat/foo"], git_repo)
+    _write_commit(git_repo, "foo.txt", "foo\n", "add foo")
+    assert resolver.resolve(cwd=git_repo) == "main"
+
+
+def test_auto_detect_main_plus_stage_branched_from_stage(git_repo: Path) -> None:
+    """stage から切った feature branch は stage を選ぶ。"""
+    # main にもう 1 コミット（stage と main が diverge するのを模す）
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)